File: src/main/java/com/mudhut/nudge/users/repositories/VerificationTokenRepository.java
package com.mudhut.nudge.users.repositories;

import org.springframework.data.jpa.repository.JpaRepository;

import com.mudhut.nudge.users.entities.VerificationToken;

import java.util.Optional;

public interface VerificationTokenRepository extends JpaRepository<VerificationToken, Long> {
    Optional<VerificationToken> findByToken(String token);
}

File: src/main/kotlin/com/mudhut/nudge/users/repositories/VerificationTokenRepository.kt
package com.mudhut.nudge.users.repositories

import com.mudhut.nudge.users.entities.VerificationToken
import org.springframework.data.jpa.repository.JpaRepository
import java.util.Optional

interface VerificationTokenRepository : JpaRepository<VerificationToken, Long> {
    fun findByToken(token: String): Optional<VerificationToken>
}
